typegate/tests/utils/test.ts

```diff
@@ -65,16 +63,17 @@ export class MetaTest {
   meta = defaultCli;
   workingDir = testDir;
   port: number | null = null;
+  currentTypegateIndex = 0;
 
   static async init(
     t: Deno.TestContext,
-    typegate: Typegate,
+    typegates: Typegate[],
     introspection: boolean,
     port = false,
   ): Promise<MetaTest> {
-    const mt = new MetaTest(t, typegate, introspection);
+    const mt = new MetaTest(t, typegates, introspection);
     if (port) {
-      const { port: p, cleanup } = await serve(typegate);
+      const { port: p, cleanup } = await serve(typegates[0]);
```

Comment:
Should't all of 'em be passed to serve?

Reply:
the `serve` function is called once during the MetaTest init and the request handler is assigned only once during the time the test is initialized. So I don't know how we can apply scheduling there. I was thinking of writing a method that returns engines from the multiple instances of typegates. Then doing scheduling(emulated) the request from the test file. i.e from `wasmedge_sync_test.ts` for instance. What do you think @Natoandro.